contracts/modules/TransactionManager.sol

```diff
@@ -275,9 +282,22 @@ abstract contract TransactionManager is BaseModule {
     }
 
     function recoverSpender(address _wallet, Call calldata _transaction) internal pure returns (address) {
-        if (_transaction.isTokenCall) {
+        bytes4 methodId;
```

Comment:
Since the data is in calldata I think we can get the methodId by doing `bytes4 methodId = abi.decode(_transaction.data[:4], (bytes4));` which might be more readable and doesn't require to store the data in memory.

Also, we should skip all the logic if `_transaction.data.length < 4` which would correspond to a ETH transfer.

Reply:
We'd need to do `bytes32 res = abi.decode(abi.encodePacked(bytes28(0), _transaction.data), (bytes32));` which costs 500 extra gas compared to the assembly code. Agreed with adding the `_transaction.data.length < 4` check though.